File: macos/Rockbox/Models/Core/Artist.swift
//
//  Artist.swift
//  Rockbox
//
//  Created by Tsiry Sandratraina on 14/12/2025.
//
import SwiftUI

struct Artist: Identifiable {
    let id = UUID()
    let cuid: String
    let name: String
    let genre: String
    let color: Color
    
    func albums(from allAlbums: [Album]) -> [Album] {
        allAlbums.filter { $0.artist == name }
    }
    
    func tracks(from allAlbums: [Album]) -> [Song] {
        albums(from: allAlbums).flatMap { $0.tracks }
    }
}

File: sdk/elixir/lib/rockbox/artist.ex
defmodule Rockbox.Artist do
  @moduledoc "An artist with their albums and tracks."

  @type t :: %__MODULE__{
          id: String.t(),
          name: String.t(),
          bio: String.t() | nil,
          image: String.t() | nil,
          tracks: [Rockbox.Track.t()],
          albums: [Rockbox.Album.t()]
        }

  defstruct [:id, :name, :bio, :image, tracks: [], albums: []]
end
